kernel/aster-nix/src/net/socket/ip/stream/mod.rs

```diff
@@ -84,8 +101,8 @@ impl StreamSocket {
             let pollee = Pollee::new(IoEvents::empty());
             init_stream.init_pollee(&pollee);
             Self {
-                options: RwLock::new(OptionSet::new()),
-                state: RwLock::new(Takeable::new(State::Init(init_stream))),
+                options: RwMutex::new(OptionSet::new()),
+                state: RwMutex::new(Takeable::new(State::Init(init_stream))),
```

Comment:
While I agree that changing this to mutex might be right (and unavoidable if we want to copy from the user buffer directly to the socket buffer), I don't see the value of changing it in this PR?

Reply:
I have removed the change from this PR.

I used to want hold the read lock when `wait_events`, just like what linux does. But since this change is not covered by this PR, the lock change should also not be included.